internal/lg/lg.go

```diff
@@ -41,35 +43,51 @@ func (l LogLevel) String() string {
 	case 5:
 		return "FATAL"
 	}
-	panic("invalid LogLevel")
+	return "invalid"
 }
 
-func ParseLogLevel(levelstr string, verbose bool) (LogLevel, error) {
-	lvl := INFO
-
+func ParseLogLevel(levelstr string) (LogLevel, error) {
 	switch strings.ToLower(levelstr) {
 	case "debug":
-		lvl = DEBUG
+		return DEBUG, nil
 	case "info":
-		lvl = INFO
+		return INFO, nil
 	case "warn":
-		lvl = WARN
+		return WARN, nil
 	case "error":
-		lvl = ERROR
+		return ERROR, nil
 	case "fatal":
-		lvl = FATAL
-	default:
-		return lvl, fmt.Errorf("invalid log-level '%s'", levelstr)
+		return FATAL, nil
 	}
-	if verbose {
-		lvl = DEBUG
+	return 0, fmt.Errorf("invalid log level '%s' (debug, info, warn, error, fatal)", levelstr)
+}
+
+type LogLevelOption LogLevel
+
+func (l *LogLevelOption) Set(s string) error {
+	lvl, err := ParseLogLevel(s)
+	if err != nil {
+		return err
 	}
-	return lvl, nil
+	*l = LogLevelOption(lvl)
+	return nil
 }
 
+func (l *LogLevelOption) Get() interface{} { return LogLevel(*l) }
+
+func (l *LogLevelOption) String() string { return LogLevel(*l).String() }
```

Comment:
just curious, why make another type, instead of just giving LogLevel these methods? is it because this `String()` needs to be for a pointer type?

Reply:
that's a damn good question